src/backend/distributed/deparser/deparse_statistics_stmts.c

```diff
@@ -0,0 +1,141 @@
+ *
+ * deparse_statistics_stmts.c
+ *	  All routines to deparse statistics statements.
+ *	  This file contains all entry points specific for statistics statement deparsing
+ *    as well as functions that are currently only used for deparsing of the statistics
+ *    statements.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+#include "postgres.h"
+
+#include "distributed/citus_ruleutils.h"
+#include "distributed/deparser.h"
+#include "distributed/relay_utility.h"
+#include "lib/stringinfo.h"
+#include "nodes/nodes.h"
+#include "utils/builtins.h"
+
+static void AppendCreateStatisticsStmt(StringInfo buf, CreateStatsStmt *stmt);
+static void AppendStatisticsName(StringInfo buf, CreateStatsStmt *stmt);
+static void AppendStatTypes(StringInfo buf, CreateStatsStmt *stmt);
+static void AppendColumnNames(StringInfo buf, CreateStatsStmt *stmt);
+static void AppendTableName(StringInfo buf, CreateStatsStmt *stmt);
+
+char *
+DeparseCreateStatisticsStmt(Node *node)
+{
+	CreateStatsStmt *stmt = castNode(CreateStatsStmt, node);
+
+	StringInfoData str;
+	initStringInfo(&str);
+
+	AppendCreateStatisticsStmt(&str, stmt);
+
+	return str.data;
+}
+
+
+static void
+AppendCreateStatisticsStmt(StringInfo buf, CreateStatsStmt *stmt)
+{
+	appendStringInfo(buf, "CREATE STATISTICS ");
+
+	appendStringInfo(buf, "%s", stmt->if_not_exists ? "IF NOT EXISTS " : "");
+
+	AppendStatisticsName(buf, stmt);
+
+	AppendStatTypes(buf, stmt);
+
+	appendStringInfo(buf, " ON ");
+
+	AppendColumnNames(buf, stmt);
+
+	appendStringInfo(buf, " FROM ");
+
+	AppendTableName(buf, stmt);
+
+	appendStringInfo(buf, ";");
+}
+
+
+static void
+AppendStatisticsName(StringInfo buf, CreateStatsStmt *stmt)
+{
+	Value *schemaNameVal = (Value *) linitial(stmt->defnames);
+	char *schemaName = strVal(schemaNameVal);
+
+	Value *statNameVal = (Value *) lsecond(stmt->defnames);
+	char *statName = strVal(statNameVal);
+
+	appendStringInfo(buf, "%s.%s", quote_identifier(schemaName), quote_identifier(
+						 statName));
+}
+
+
+static void
+AppendStatTypes(StringInfo buf, CreateStatsStmt *stmt)
+{
+	if (list_length(stmt->stat_types) == 0)
+	{
+		return;
+	}
+
+	appendStringInfo(buf, " (");
+
+	ListCell *cell = NULL;
+	foreach(cell, stmt->stat_types)
+	{
+		Value *statType = (Value *) lfirst(cell);
+
+		appendStringInfoString(buf, strVal(statType));
+
+		if (cell != list_tail(stmt->stat_types))
+		{
+			appendStringInfo(buf, ", ");
+		}
+	}
+
+	appendStringInfo(buf, ")");
+}
+
+
+static void
+AppendColumnNames(StringInfo buf, CreateStatsStmt *stmt)
+{
+	ListCell *cell = NULL;
+	foreach(cell, stmt->exprs)
+	{
+		Node *node = (Node *) lfirst(cell);
+		Assert(IsA(node, ColumnRef));
+
+		ColumnRef *column = (ColumnRef *) node;
+		Assert(list_length(column->fields) == 1);
+
+		char *columnName = strVal((Value *) linitial(column->fields));
+
+		appendStringInfoString(buf, columnName);
```

Comment:
```sql
[local] nilsdijk@postgres:9700-61144=# CREATE TABLE foo(a int, "Bar" text);
CREATE TABLE
Time: 47.247 ms

[local] nilsdijk@postgres:9700-61144=# SELECT create_distributed_table('foo', 'a');
┌──────────────────────────┐
│ create_distributed_table │
├──────────────────────────┤
│                          │
└──────────────────────────┘
(1 row)

Time: 51.515 ms

[local] nilsdijk@postgres:9700-61144=# CREATE STATISTICS bar ON a, "Bar" FROM foo;
ERROR:  column "bar" does not exist
CONTEXT:  while executing command on localhost:9701
Time: 12.204 ms
```

As it seems we need quoting here!
This will be solved when using `NameListToQuotedString` here.